config/routes.rb

```diff
@@ -1,4 +1,14 @@
+class AuthConstraint
+  def matches?(request)
+    request.session[:user]["email"].present?
```

Comment:
That would work. Another thought I had was that we could add a `session_token` column to the users table. When a user signs in we can set their `session_token` using `SecureRandom.urlsafe_base64(16)`. We could then update the `current_user` method like so: `	@current_user ||= User.find_by(session_token: session[:session_token])` . On sign out the session token would be reset to nil. This might be more secure than using an email, let me know!